types/ember-qunit/index.d.ts

```diff
@@ -156,7 +156,34 @@ declare module 'ember-qunit' {
 declare module 'qunit' {
     import { TestContext } from "ember-test-helpers";
 
-    export const module: typeof QUnit.module;
+    type moduleFunc1 = (name: string, hooks?: Hooks, nested?: (hooks: NestedHooks) => void) => void;
+    type moduleFunc2 = (name: string, nested?: (hooks: NestedHooks) => void) => void;
+    interface ModuleOnly { only: moduleFunc1 & moduleFunc2; }
+
+    export const module: moduleFunc1 & moduleFunc2 & ModuleOnly;
```

Comment:
This is not an unreasonable way of defining these, but it definitely took me a minute to see what you were doing: supplying overloads like this is pretty unusual. A more idiomatic way of writing this interface would look like this: 

```ts
interface ModuleFn {
  (name: string, hooks?: Hooks, nested?: (hooks: NestedHooks) => void) => void;
  (name: string, nested?: (hooks: NestedHooks) => void) => void;
}

interface Module extends ModuleFn {
  only: ModuleFn;
}

export const module: Module;
```

Both ways *work*, but this makes it clearer that we're defining a shared function interface which the `module` itself *and* the module's embedded `only` type share, and defining the `module` type as including the merge of the function type and the `only` property.

Reply:
This is borrowed from QUnit types, and I will admit I don't fully understand what it's all about.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1b51c6dafe512ceceaf5686a77532d5521960335/types/qunit/index.d.ts#L372-L374

Reply:
That's fair! Let's go ahead and make the change here, and I'll make a note to upstream it into the QUnit types as well.

By way of explanation: this exists to provide an overload, since you can call it with only `Hooks`, or only `NestedHooks`, or `Hooks` *and* `NestedHooks`—but the exact same signature applies to `module(...)` and to `only(...)`. Doing it the way I have gets you those same benefits but is a bit clearer and more maintainable. Let me know if you have further questions!

Also, does QUnit itself have docs on the `module.only` invocation? I didn't see them on the Ember side, but I also didn't dive that far into it!

Reply:
The recent refactor I did in #47731 should make it unnecessary to include this at all here.